_includes/figure

```diff
@@ -1,17 +1,82 @@
+{% capture this-figure %}
 {% comment %}
 To do: create HTML5 <figure> option here based on site.output
-when theme stylesheet has definitions. Problem is that Jekyll/kramdown 
-adds paragraph tags around the include.foo arguments. 
 {% endcomment %}
 
-> ![{{ include.description }}]({{ path-to-book-directory }}{{ site.image-set}}/{{ include.image }})
-> {% if include.image-height != nil %}{:.height-{{ include.image-height }}}{% endif %}
->
-> {% if include.reference != nil %}*{{ include.reference }}*{:.figure-reference} {% endif %}{{ include.caption }}
-> {:.caption}
-{:.figure}
+{% include metadata %}
 
-{% if include.source != nil %}
-{{ include.source }}
-{:.source }
+{% comment %}
+Get the list of images, whether included as `image` or `images`
+and remove spaces that might be around commas.
+Put the comma-separated images into an array.
+Then get the total number of images in the array.
+{% endcomment %}
+{% capture image-list %}{{ include.image | remove: " " }}{% if include.image and include.images %},{% endif %}{{ include.images | remove: " " }}{% endcapture %}
+{% assign images = image-list | split: "," %}
+{% assign number-of-images = images | size %}
+
+{% comment %}
+Then create our figure, using multiple images if necessary.
+This uses a blockquote element to be XHTML-1.1 compliant for EPUB2.
+We have to do funky remove: '<p>' | remove: '</p>' | strip_newlines
+because https://github.com/jekyll/jekyll/issues/2248.
+{% endcomment %}
+
+<blockquote class="figure{% if include.class %} {{ include.class }}{% endif %}"{% if include.reference %} id="{{ include.reference | slugify }}"{% else %} id="{{ images[0] | replace: ".", " " | truncatewords: 1, "" }}"{% endif %}>
+
+<div class="figure-body">
+
+<div class="figure-images contains-{{ number-of-images }}{% if include.image-height != nil %} height-{{ include.image-height }}{% endif %}">
```

Comment:
@SteveBarnett Good question, but the problem is not the class but where I've applied it here, I think.

The `height` class is for print: sometimes you want to force an image to be a specific number of lines high, so that the page's baseline grid is consistent (i.e. all lines match a grid defined by the default line height, e.g. 15pt). An image that is, say 10.3 lines high will throw out a page's baseline grid by 0.3 lines. Our template Sass partials support a `height-x` class [here](https://github.com/electricbookworks/electric-book/blob/master/_sass/partials/_print-fitting.scss). 

But we should be applying that class to the `img`, right, not the `div`, right? I'll put that in a PR.